### ISSUE TITLE
Update the "Send a bug report" window to use GitHub

### ISSUE THREAD

**Issue by jsmaniac** [racket/drracket#114]
The "Send a bug report" window uses http://bugs.racket-lang.org/, which seems to be the old bug tracker (and GitHub is preferred nowadays).

The submit code should be updated to use GitHub instead.

It would be possible to either open `https://github.com/racket/REPO/issues/new?title=foo&body=bar` in the user's browser, or use the GitHub API to directly submit the bug (this would however likely require that the user has a GitHub account).

**Comment by bennn** [comment]
> (this would however likely require that the user has a GitHub account)

I think we could make a "DrRacket User" account on GitHub, and Dr.Racket could store whatever auth keys GitHub requires.

**Comment by jsmaniac** [comment]
Wouldn't this have a high risk of being hijacked and the account misused?

I'd say disclosing the account's OAuth token or other API key would be unwise, unless GitHub's permission system is fine-grained enough to get a token which only allows submitting bugs for a small set of repositories.

We could however have a minimal server which just accepts a title and body string, and posts it as an issue on one of the allowed repos on the user's behalf. The difference is that keys would be stored on the server and not disclosed. I'm just making guesses here, though.

**Comment by rfindler** [comment]
I think that something inside DrRacket that directs someone to a relevant github page (and maybe a FAQ page about issues that we make on our wiki?) is sufficient.

**Comment by jsmaniac** [comment]
GitHub's issue templates can also be used if we want to make sure some information is filled in by the template (like is done for the TR repo).

**Comment by samth** [comment]
On Wed, May 17, 2017, 7:32 PM Georges Dupéron <notifications@github.com>
wrote:

> Wouldn't this have a high risk of being hijacked and the account misused?
>
Yes.

> I'd say disclosing the account's OAuth token or other API key would be
> unwise, unless GitHub's permission system is fine-grained enough to get a
> token which only allows submitting bugs for a small set of repositories.
>
This isn't possible, and embedding the key is not allowed by GitHub's
policies.

> We could however have a minimal server which just accepts a title and body
> string, and posts it as an issue on one of the allowed repos on the user's
> behalf. The difference is that keys would be stored on the server and not
> disclosed. I'm just making guesses here, though.
>
We could do that, but I don't want to run a server for just that.

**Comment by LeifAndersen** [comment]
> I think that something inside DrRacket that directs someone to a relevant github page (and maybe a FAQ page about issues that we make on our wiki?) is sufficient.

How many people using the submit a bug tool actually have github accounts though? And creating one may be a large barrier to entry for new users to submit a bug.

**Comment by rfindler** [comment]
Yes, I agree this a concern. When compared with the alternatives, however, I think it is the best option we have. 

The wiki page could also suggest posting on our mailing list, which has a lower barrier to entry, and helpful people who would, I expect, either submit issues or explain how to.

**Comment by LeifAndersen** [comment]
Fair concern. What about using something like this?: https://gitreports.com/

Then we can have github issues without managing our own server or requiring users to log in.

**Comment by jsmaniac** [comment]
@LeifAndersen https://gitreports.com/ looks like a great solution. Two minor itches:

* The captcha is not accessible to ill-sighted & blind people _(sigh)_
* gitreports needs full write access to all the user or organisation's repo. Given the perks (possibility to inject code into Rakcet and other projects), gitreports seems like a reasonably attractive target, so it makes me a bit weary. Granting gitreports access to an isues-only repo (in a separate organisation, say racket-issues) seems an option, but as @samth said on IRC, having an issues-only repo might not be a good fit for Racket.

**Comment by LeifAndersen** [comment]
Regarding your first point...ugg...I didn't notice the captcha. I'll file a bug report with that repo, they realliy should have a low vision option.

**Comment by spdegabrielle** [comment]
Hi,

 I believe the current situation is worse than users having to create a GitHub account.  

While a perfect solution is being developed, I have created an interim solution using the default 'submit bug report' template.

This required two actions
a) a bug report template for racket/racket. See PR for proposed template https://github.com/racket/racket/pull/3031
b) change menu in DrRacket to link to new bug report e.g. https://github.com/spdegabrielle/racket/issues/new?assignees=&labels=bug&template=bug_report.md&title=%5BBUG%5D+

As there are already existing examples of this I believe (b) could be a `good first issue` for someone interested in DrRacket .

I note that the title is passed as a parameter - this is worth investigating to see if some environment details can be passed this way, as per the old bug report mechanism.

Kind regards,

Stephen

----

Current situation if user attempts to us the submit bug via the help screen; 

User fills in form, presses submit, after a long wait the following error is returned;

> 
> ERROR:
> tcp-connect: connection failed
>   address: bugs.racket-lang.org
>   port number: 80
>   system error: Operation timed out; errno=60
>   (http-conn-open!10 . #(struct:srcloc #pplications/Racket v7.5/collects/net/http-client.rkt> 67 0 1446 2166))
>   (#f . #(struct:srcloc #pplications/Racket v7.5/collects/net/http-client.rkt> 272 0 9123 269))
>   (#f . #(struct:srcloc #pplications/Racket v7.5/collects/racket/contract/private/arrow-val-first.rkt> 555 3 23822 3167))
>   (http://getpost-impure-port . #(struct:srcloc #pplications/Racket v7.5/collects/net/url.rkt> 201 0 7433 1092))
>   (getpost-pure-port . #(struct:srcloc #pplications/Racket v7.5/collects/net/url.rkt> 271 0 10246 1055))
>   (#f . #(struct:srcloc #pplications/Racket v7.5/collects/racket/contract/private/arrow-val-first.rkt> 555 3 23822 3167))
>   (.../more-scheme.rkt:261:28 . #f)
> 
>
